bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -51,80 +60,168 @@
 
     private @Nullable GardenaDeviceDiscoveryService discoveryService;
 
     private @Nullable GardenaSmart gardenaSmart;
     private HttpClientFactory httpClientFactory;
     private WebSocketFactory webSocketFactory;
 
+    private final Object reInitializationCodeLock = new Object();
+    private @Nullable ScheduledFuture<?> reInitializationTask;
+    private boolean reInitializationCausedBy429 = false;
+    private Instant lastApiCallTime = Instant.MIN;
+    private boolean lastApiCallTimeLoaded = false;
+
     public GardenaAccountHandler(Bridge bridge, HttpClientFactory httpClientFactory,
             WebSocketFactory webSocketFactory) {
         super(bridge);
         this.httpClientFactory = httpClientFactory;
         this.webSocketFactory = webSocketFactory;
     }
 
+    private Instant lastApiCallTime() {
+        if (!lastApiCallTimeLoaded) {
+            Map<String, String> properties = getThing().getProperties();
+            String property = properties.getOrDefault(GardenaBindingConstants.LAST_API_CALL_TIME, "");
+            lastApiCallTime = "".equals(property) ? Instant.now().minus(Duration.ofHours(1)) : Instant.parse(property);
+            lastApiCallTimeLoaded = true;
+        }
+        return lastApiCallTime;
+    }
+
+    private void lastApiCallTimeUpdate() {
+        lastApiCallTime = Instant.now();
+        getThing().setProperty(GardenaBindingConstants.LAST_API_CALL_TIME,
+                lastApiCallTime.truncatedTo(ChronoUnit.SECONDS).toString());
+    }
+
     @Override
     public void initialize() {
         logger.debug("Initializing Gardena account '{}'", getThing().getUID().getId());
-        initializeGardena();
+        Instant now = Instant.now();
+        Instant notBeforeTime = lastApiCallTime().plus(REINITIALIZE_DELAY_MINUTES_BACK_OFF)
+                .plus(REINITIALIZE_DELAY_MINUTES_BACK_OFF);
+        if (now.isBefore(notBeforeTime)) {
+            // delay the initialisation
+            Duration delay = Duration.between(now, notBeforeTime);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, uiText(delay.getSeconds()));
+            scheduleReinitialize(delay);
+        } else {
+            // do immediate initialisation
+            scheduler.submit(() -> initializeGardena());
+        }
     }
 
     public void setDiscoveryService(GardenaDeviceDiscoveryService discoveryService) {
         this.discoveryService = discoveryService;
     }
 
+    /**
+     * Format a description text to display on the thing main UI page
+     *
+     * @param delaySeconds the delay that will be added to the current time
+     * @return the description text
+     */
+    public String uiText(long delaySeconds) {
+        return "Waiting to make automatic reconnection attempt at " + LocalDateTime.now().plusSeconds(delaySeconds)
+                .truncatedTo(ChronoUnit.SECONDS).format(DateTimeFormatter.ofPattern("HH:MM:SS (cccc)"));
```

Comment:
If this is an UI text, shouldn't it be localized?

Reply:
Yes. But I don't recall how to do that.

Reply:
Using [`TranslationProvider`](https://www.openhab.org/docs/developer/utils/i18n.html#i18n-text-provider-api). In your case, the string would be something like `gardena.delayed-reconnection-status = Waiting to make automatic reconnection attempt at {0}`, and the code to generate the string something like

```
DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedTime(FormatStyle.SHORT);
String delayedTimestamp = LocalDateTiume.now().plusSeconds(delaySeconds).format(formatter);
return i18nProvider.getText(bundle, "gardena.delayed-reconnection-status", "", localeProvider.getLocale(), delayedTimestamp);
```

i18nProvider is the `TranslationProvider` instance and `localeProvider` a `LocaleProvider` instance, both of which need to be fetched as OSGi services in the `HandlerFactory`.

Reply:
I figured it out. Done.